arrow-data-source/common/src/main/scala/org/apache/spark/sql/execution/datasources/v2/arrow/SparkVectorUtils.scala

```diff
@@ -81,6 +81,18 @@ object SparkVectorUtils {
     }
   }
 
+  def setNumRows(batch: ColumnarBatch, numRows: Int): Unit = {
+    batch.setNumRows(numRows)
+    for (i <- 0 until batch.numCols()) {
+      val vector = batch.column(i)
+      vector match {
+        case av: ArrowWritableColumnVector =>
+          av.setValueCount(numRows)
+        case _ =>
+      }
+    }
+  }
+
```

Comment:
We'd better have a customized `ColumnarBatch` impl to override `ColumnarBatch#setNumRows` however it's not allowed because of `ColumnarBatch`'s `final` modifier.